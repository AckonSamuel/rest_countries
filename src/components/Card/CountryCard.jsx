import PropTypes from 'prop-types';
import { useNavigate } from 'react-router-dom';
import Card from '@mui/material/Card';
import CardContent from '@mui/material/CardContent';
import CardMedia from '@mui/material/CardMedia';
import Typography from '@mui/material/Typography';
import CardActionArea from '@mui/material/CardActionArea';
import pxToRem from '../../assets/theme/pxToRem';

const CountryCard = ({
    officialName,
    region,
    capital,
    flagsvg,
    flagsalt,
    population,
}) => {
    const navigate = useNavigate();

    const goTo = (sm) => {
        navigate(sm);
    };

    return (
        <Card sx={{ width: pxToRem(264), height: pxToRem(336) }}>
            <CardActionArea>
                <CardMedia
                    component="img"
                    sx={{ height: pxToRem(160) }}
                    width="100%"
                    image={flagsvg}
                    alt={flagsalt}
                    onClick={() => goTo(`/details/${officialName}`)}
                />
                <CardContent 
                sx={{ height: pxToRem(176) }}
                >
                    <Typography gutterBottom variant="h5" component="div">
                        {officialName}
                    </Typography>
                    <Typography variant="body2" color="text.secondary">
                        Population: {population} <br />
                        Region: {region} <br />
                        Capital: {capital}
                    </Typography>
                </CardContent>
            </CardActionArea>
        </Card>
    )
};

CountryCard.propTypes = {
    officialName: PropTypes.string.isRequired,
    flagsalt: PropTypes.string.isRequired,
    flagsvg: PropTypes.string.isRequired,
    region: PropTypes.string.isRequired,
    population: PropTypes.number.isRequired,
    // capital: PropTypes.string.isRequired,
};

export default CountryCard;