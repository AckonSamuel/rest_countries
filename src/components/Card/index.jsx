import PropTypes from 'prop-types';
import CountryCard from './CountryCard';

const CountryList = ({ countries, searchText, filteredSearch }) => {
  if (filteredSearch && filteredSearch.length === 0 && searchText.length !== 0) {
    return <h1 className="error-msg">Oops! Search not found</h1>;
  }
  return (
    <section id="country-list">

      { countries && countries.length && countries.map((country) => (
        <CountryCard
          key={country.name.official}
          officialName={country.name.common}
          flagsalt={country.flags.alt}
          flagsvg={country.flags.svg}
          region={country.region}
          capital={country.capital}
          population={country.population}
        />
      ))}
    </section>
  );
};

CountryList.propTypes = {
  countries: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.any)).isRequired,
  filteredSearch: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.any)).isRequired,
  searchText: PropTypes.string.isRequired,
};

export default CountryList;
