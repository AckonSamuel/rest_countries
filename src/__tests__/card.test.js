import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Card from '../components/Card';
import countries from '../assets/testdata';

it('renders correctly', () => {
  const filteredSearch = [];
  const searchText = '';

  const tree = renderer
    .create(
      <BrowserRouter>
        <Card countries={countries} filteredSearch={filteredSearch} searchText={searchText} />
        ,
      </BrowserRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
