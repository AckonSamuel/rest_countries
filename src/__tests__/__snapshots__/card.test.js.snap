// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <section
    id="country-list"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1brhw7e-MuiPaper-root-MuiCard-root"
    >
      <button
        className="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <img
          alt="The flag of Marshall Islands has a blue field with two broadening adjacent diagonal bands of orange and white that extend from the lower hoist-side corner to the upper fly-side corner of the field. A large white star with twenty-four rays — four large rays at the cardinal points and twenty smaller rays — is situated in the upper hoist-side corner above the diagonal bands."
          className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-ta2yni-MuiCardMedia-root"
          onClick={[Function]}
          src="https://flagcdn.com/mh.svg"
          width="100%"
        />
        <div
          className="MuiCardContent-root css-nw3cq4-MuiCardContent-root"
        >
          <div
            className="MuiTypography-root MuiTypography-cardTitle MuiTypography-gutterBottom css-anmq6p-MuiTypography-root"
          >
            Marshall Islands
          </div>
          <span
            className="MuiTypography-root MuiTypography-cardContent css-37jdci-MuiTypography-root"
          >
            <span
              style={
                Object {
                  "fontWeight": 600,
                }
              }
            >
              Population:
            </span>
             
            59,194
             
            <br />
            <span
              style={
                Object {
                  "fontWeight": 600,
                }
              }
            >
              Region: 
            </span>
             
            Oceania
             
            <br />
            <span
              style={
                Object {
                  "fontWeight": 600,
                }
              }
            >
              Capital:
            </span>
             
            Majuro
          </span>
        </div>
        <span
          className="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
        />
      </button>
    </div>
  </section>,
  ",",
]
`;
